shortlister/model.py

```diff
@@ -90,18 +98,59 @@ def load_role(path, criteria):
 
 def load_applicants(path: Path):
     """Generate a list of applicant instances from pdf format CVs."""
-    files = path.glob("*.pdf")
+    files = list(path.glob("*.pdf"))
     applicants = []
     for file in files:
-        name_parts = file.stem.split("_")
-        applicant = Applicant(
-            name=" ".join(name_parts[0:2]), cv=file, scores={}, notes=""
-        )
-        applicants.append(applicant)
+        try:
+            applicant = load_applicants_from_pdf(Path(file))
+            applicants.append(applicant)
+        except ValueError:
+            print(f"Corrupt/incorrect pdf file: {file}")
+            continue
     sort_alpha(applicants)
     return applicants
 
 
+def load_applicants_from_pdf(file: Path):
+    """Create single Applicant instance from PDF files in the role directory"""
+    doc = pymupdf.open(file)
+    # takes the first page of the pdf (the candidate pack)
+    page = doc[0]
+    # extract text in reading order
+    text = page.get_text(sort=True)
+    # turns text into a list of string representing each extracted line
+    lines = text.splitlines()
+    # remove empty line from list
+    cleaned = [line.strip() for line in lines if len(line.strip())]
+
+    # sets the value of each field
+    info = extract_info_from_text(cleaned)
+    (
+        first_name,
+        last_name,
+        email,
+        phone,
+        postcode,
+        country_region,
+        applicant_right_to_work,
+        visa_req_text,
+    ) = [i for i in info]
+    # create applicant instance with above information
+    applicant = Applicant(
+        name=f"{first_name} {last_name}",
```

Comment:
If we weren't able to extract the name from the PDF text, get it from the name of the PDF file as you were (see my comments on `extract_info_from_text` function below)